CLR/Libraries/Windows_Devices/windows_devices_native_Windows_Devices_Gpio_GpioPin.cpp

```diff
@@ -0,0 +1,234 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "windows_devices.h"
+
+enum GpioPinDriveMode
+{
+    GpioPinDriveMode_Input = 0,
+    GpioPinDriveMode_Output,
+    GpioPinDriveMode_InputPullUp,
+    GpioPinDriveMode_InputPullDown,
+    GpioPinDriveMode_OutputStrongLow,
+    GpioPinDriveMode_OutputStrongLowPullUp,
+    GpioPinDriveMode_OutputStrongHigh,
+    GpioPinDriveMode_OutputStrongHighPullDown,
+};
+
+enum GpioPinValue
+{
+    GpioPinValue_Low = 0,
+    GpioPinValue_High,
+};
+
+static GPIO_RESISTOR GetResisterMode(GpioPinDriveMode driveMode)
+{
+    switch (driveMode)
+    {
+    case GpioPinDriveMode_InputPullUp:
+    case GpioPinDriveMode_OutputStrongLowPullUp:
+        return RESISTOR_PULLUP;
+
+    case GpioPinDriveMode_InputPullDown:
+    case GpioPinDriveMode_OutputStrongHighPullDown:
+        return RESISTOR_PULLDOWN;
+    }
+
+    return RESISTOR_DISABLED;
+}
+
+void IsrProcedure( GPIO_PIN pin, BOOL pinState, void* context )
+{
+    ASSERT_IRQ_MUST_BE_OFF();
+    PostManagedEvent( EVENT_GPIO, 0, pin, pinState );
+}
+
+HRESULT Library_windows_devices_native_Windows_Devices_Gpio_GpioPin::get_DebounceTimeout___mscorlibSystemTimeSpan( CLR_RT_StackFrame& stack )
+{
+    TINYCLR_HEADER();
+
+    CLR_INT64 value;
+    CLR_RT_HeapBlock* pThis = stack.This(); FAULT_ON_NULL(pThis);
+
+    if (pThis[ FIELD__m_disposed ].NumericByRef().s1 != 0)
+    {
+        TINYCLR_SET_AND_LEAVE(CLR_E_OBJECT_DISPOSED);
+    }
+
+    value = (CLR_INT64)::CPU_GPIO_GetDebounce() * TIME_CONVERSION__TO_MILLISECONDS;
+    stack.SetResult_I8( value );
+    stack.TopValue().ChangeDataType( DATATYPE_TIMESPAN );
+
+    TINYCLR_NOCLEANUP();
+}
+
+HRESULT Library_windows_devices_native_Windows_Devices_Gpio_GpioPin::set_DebounceTimeout___VOID__mscorlibSystemTimeSpan( CLR_RT_StackFrame& stack )
+{
+    TINYCLR_HEADER();
+
+    CLR_INT64 value;
+    CLR_RT_HeapBlock* pThis = stack.This(); FAULT_ON_NULL(pThis);
+
+    if (pThis[ FIELD__m_disposed ].NumericByRef().s1 != 0)
+    {
+        TINYCLR_SET_AND_LEAVE(CLR_E_OBJECT_DISPOSED);
+    }
+
+    value = (CLR_INT64_TEMP_CAST) stack.Arg1().NumericByRef().s8 / TIME_CONVERSION__TO_MILLISECONDS;
+
+    // TODO: Set debounce individually per pin once the HAL has that capability.

```

Comment:
What's the plan to get this implemented? As a general rule code being checked in should not contain "todo:" or any other similar variant of comment indicating known work that isn't completed. Unless they explicitly list an item id for the issue in the issue tracking system governing the code. Consider/review/Future type comments that hint at alternate approaches or capabilities in a future release are OK.